infrastructure/src/main/kotlin/DetektPlugin.kt

```diff
@@ -63,6 +50,7 @@ private inline fun Project.detektTask(
         configure()
         parallel = true
         config.setFrom(extension.configsDir.file("detekt/detekt.yml"))
+        baseline.set(extension.configsDir.file("detekt/baseline.xml"))
```

Comment:
Maybe before installing, need to check if the file exists? Otherwise Detekt tasks fail with a file missing error.

Reply:
Hmm, I thought this behavior was fixed in Gradle 1.17.0 - https://github.com/detekt/detekt/pull/3619

Reply:
Can you provide the stacktrace?

Reply:
Type 'Detekt' property 'baseline' specifies file '/.../config/detekt/baseline.xml' which doesn't exist.
    
    Reason: An input file was expected to be present but it doesn't exist.
    
    Possible solutions:
      1. Make sure the file exists before the task is called.
      2. Make sure that the task which produces the file is declared as an input.
    
    Please refer to https://docs.gradle.org/7.0/userguide/validation_problems.html#input_file_does_not_exist for more details about this problem.

Reply:
The stack trace does not provide any detekt classes and only contains inner classes. You can reproduce this case by deleting the baseline in the project.

Reply:
I have researched this behavior. Detekt's implementation is correct, but when you set `Provider` to `Property`, gradle will check the `Provider`, for example if it is a file, Gradle will check if the file exists. 

You can find it in `org.gradle.api.internal.tasks.properties.ValidationActions`.

One possible solution is to create a configFile property in RedmadrobotExtension, and then it will work correctly.

Reply:
Thank you for this research!

Reply:
Created issue #52 to continue discussion